students/CarolF/session04/MailRoom_2.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env Python3
+
+################################
+##Created By: Carol Farris
+##Date: February 13th 2018
+##Goal: MailRoom Part 2 Assignment. 
+##Changes: Swapped lists with dictionaries where it was best to do so. Added additional option to write letter for all donors to local file.
+################################
+
+
+def getAction(usersChoice):
+    """This method will prompt the user for a specific action: Send thank you, create report or quit, then route it to correct method""" 
+    ##put the four items in a dictionary
+    
+    for key, value in usersChoice.items():    print('%s: %s' % (key, value))
+
+    response = int(input("Please type a numeric selection from the above three actions >"))	
+
+    while response != 1 and response != 2 and response != 3 and response != 4:
+        print("Invalid entry. Expected a '1', '2' or '3' ")
+        for key, value in usersChoice.items():    print('%s: %s' % (key, value))
+        response = int(input("Please type a numeric selection from the above three actions > "))
+
+    if response ==4:
+        print("Goodbye")
+        exit() 
+
+    if response ==3:
+        print("Send Letters to All")
+        sendLettersToAll()      
+
+    if response ==2:
+        createReportData()	
+
+    if response == 1: 
+        findDonorName()
+
+def findDonorName():
+    """This function identifies the donor information to be used"""
+    donorName = str(input("Please provide the donor name for the thank you or type 'list' to see the list of donors >"))
+
+    while donorName == 'list': #as long as user types list, it will return the donor list
+        for key in donorList:  print(key) #dictionary version to provide list
+        donorName = str(input("Please enter the donor name to compose the thank you >"))
+    
+    if(donorName not in donorList):
+        isNewDonor = True 
+        print(f'Adding {donorName} as new donor to list') #remove after debugging
+        retrieveDonationAmount(donorName, isNewDonor)
+        
+    elif (donorName in donorList):
+        isNewDonor = False
+        print("found! ", donorName)
+        retrieveDonationAmount(donorName, isNewDonor)
```

Comment:
Note that `donorName in donorList` will give you a boolean, so you can just send that in: `retrieveDonationAmount(donorName, donorName in donorList)`